Ex4/Ex4.java

```diff
@@ -0,0 +1,81 @@
+public class Ex4 {
+
+	public static final boolean LAND = true;
+	public static final boolean WATER = false;
+	
+	public static boolean[][] map = {{WATER, LAND, WATER, LAND},
+					 {LAND, LAND, WATER, WATER},
+					 {WATER, WATER, LAND, WATER},
+					 {WATER, WATER, LAND, WATER}};
+	
+	public static void main(String[] args) {
+		System.out.print("Number of islands in the map: ");
```

Comment:
It would be nice to have more lower level tests. Eg:

testTopLeftCorner()
testNonSquareMap()
testWholeMapIsland()
testNoIslands()
testUShape()
testDiagonalIsntCOnnected().

If you have never tried it, I'd recommend learning junit. It is used by almost every company in the world who write Java. 
http://junit.org/junit4/